File: POLIMORFISMO/EJERCICIO 1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="Desconocido", plataforma="PC", cantidadJugadores=0):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = cantidadJugadores

    def mostrar(self):
        print(self.__str__())

    def agregarJugadores(self, cantidad=1):
        self.cantidadJugadores += cantidad

    def __str__(self):
        return f"Videojuego - Nombre: {self.nombre}, Plataforma: {self.plataforma}, Jugadores: {self.cantidadJugadores}"


juego1 = Videojuego()
juego2 = Videojuego("FIFA 25", "PlayStation", 4)

juego1.mostrar()
juego2.mostrar()

juego1.agregarJugadores()     # suma 1
juego2.agregarJugadores(3)    # suma 3

print(juego1)
print(juego2)

File: POLIMORFISMO/EJERCICIO 1/Videojuego.java
package EJERCICIO1;

public class Videojuego {
    public String nombre;
    public String plataforma;
    public int cantidadJugadores;

    public Videojuego() {
        this.nombre = "Desconocido";
        this.plataforma = "PC";
        this.cantidadJugadores = 0;
    }

    // Constructor sobrecargado
    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    public void mostrar() {
        System.out.println(this.toString());
    }

    //Agregar 1 jugador
    public void agregarJugadores() {
        this.cantidadJugadores += 1;
    }

    //Agregar una cantidad específica de jugadores
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
    }

    public String toString() {
        return "Videojuego - Nombre: " + this.nombre +
               ", Plataforma: " + this.plataforma +
               ", Jugadores: " + this.cantidadJugadores;
    }
}
